apps/hash-graph/lib/graph/src/api/rest/entity_type.rs

```diff
@@ -116,6 +152,168 @@ struct CreateEntityTypeRequest {
+        .acquire()
+        .await
+        .map_err(report_to_response)?;
+
+    let (entity_types, operations): (Vec<_>, Vec<_>) = relationships
+        .0
+        .into_iter()
+        .map(|request| {
+            let resource = EntityTypeId::from_url(&request.resource);
+            (
+                resource,
+                (request.operation, resource, request.relation_and_subject),
+            )
+        })
+        .unzip();
+
+    let (permissions, _zookie) = authorization_api
+        .check_entity_types_permission(
+            actor_id,
+            EntityTypePermission::Update,
+            entity_types,
+            Consistency::FullyConsistent,
+        )
+        .await
+        .map_err(report_to_response)?;
+
+    let mut failed = false;
+    // TODO: Change interface for `check_entity_types_permission` to avoid this loop
+    for (_entity_type_id, has_permission) in permissions {
+        if !has_permission {
+            tracing::error!("Insufficient permissions to modify relationship for entity type");
+            failed = true;
+        }
+    }
+
+    if failed {
+        return Err(report_to_response(
+            Report::new(PermissionAssertion).attach(hash_status::StatusCode::PermissionDenied),
+        ));
+    }
+
+    // for request in relationships.0 {
+    authorization_api
+        .modify_entity_type_relations(operations)
+        .await
+        .map_err(report_to_response)?;
+
+    Ok(StatusCode::NO_CONTENT)
+}
+
+#[utoipa::path(
+    get,
+    path = "/entity-types/{entity_type_id}/relationships",
+    tag = "EntityType",
+    params(
+        ("X-Authenticated-User-Actor-Id" = AccountId, Header, description = "The ID of the actor which is used to authorize the request"),
+        ("entity_type_id" = VersionedUrl, Path, description = "The Entity type to read the relations for"),
+    ),
+    responses(
+        (status = 200, description = "The relations of the entity type", body = [EntityTypeRelationAndSubject]),
+
+        (status = 403, description = "Permission denied"),
+    )
+)]
+#[tracing::instrument(level = "info", skip(authorization_api_pool))]
+async fn get_entity_type_authorization_relationships<A>(
+    AuthenticatedUserHeader(actor_id): AuthenticatedUserHeader,
+    Path(entity_type_id): Path<VersionedUrl>,
+    authorization_api_pool: Extension<Arc<A>>,
+) -> Result<Json<Vec<EntityTypeRelationAndSubject>>, Response>
+where
+    A: AuthorizationApiPool + Send + Sync,
+{
+    let authorization_api = authorization_api_pool
+        .acquire()
+        .await
+        .map_err(report_to_response)?;
+
+    Ok(Json(
+        authorization_api
+            .get_entity_type_relations(
+                EntityTypeId::from_url(&entity_type_id),
+                Consistency::FullyConsistent,
+            )
+            .await
+            .map_err(report_to_response)?,
+    ))
+}
+
+#[utoipa::path(
+    get,
+    path = "/entity-types/{entity_type_id}/permissions/{permission}",
```

Comment:
It feels a bit weird to have a whole URL as a path segment of another URL, because we'll end up with things like the below, but I can't think of any actual problems these will introduce
`/entity-types/https://app.hash.ai/@hash/types/entity-type/permissions/v/1/permissions/edit`

Reply:
Those URLs are supposed to be encoded. Every client should do this automatically. As these are abstracted away and we don't treat the API as public anyway I don't really see an issue here.